test/contracts/meta-oracles/driftlessHistoricalPriceFeed.spec.ts

```diff
@@ -0,0 +1,462 @@
+
+        const actualTimestamp = await historicalPriceFeed.nextAvailableUpdate.callAsync();
+        const expectedTimestamp = previousTimestamp.add(ONE_DAY_IN_SECONDS);
+        expect(actualTimestamp).to.be.bignumber.equal(expectedTimestamp);
+      });
+    });
+
+    describe('when not enough time has passed to update', async () => {
+      beforeEach(async () => {
+        subjectTimeFastForward = new BigNumber(1);
+      });
+
+      it('should revert', async () => {
+        await expectRevertError(subject());
+      });
+    });
+  });
+
+  describe('#read', async () => {
+    let ethPrice: BigNumber;
+    let updatedPrices: BigNumber[];
+
+    let subjectDataDays: BigNumber;
+
+    beforeEach(async () => {
+      ethPrice = ether(150);
+      await oracleWrapper.updateMedianizerPriceAsync(
+        ethMedianizer,
+        ethPrice,
+        SetTestUtils.generateTimestamp(1000),
+      );
+
+      const updateFrequency = ONE_DAY_IN_SECONDS;
+      const updateTolerance = updateFrequency.div(4);
+      const medianizerAddress = ethMedianizer.address;
+      const dataDescription = '200DailyETHPrice';
+      const seededValues = [];
+      historicalPriceFeed = await oracleWrapper.deployDriftlessHistoricalPriceFeedAsync(
+        updateFrequency,
+        updateTolerance,
+        medianizerAddress,
+        dataDescription,
+        seededValues,
+      );
+
+      updatedPrices = await oracleWrapper.batchUpdateDriftlessHistoricalPriceFeedAsync(
+        historicalPriceFeed,
+        ethMedianizer,
+        20,
+      );
+
+      subjectDataDays = new BigNumber(20);
+    });
+
+    async function subject(): Promise<BigNumber[]> {
+      return historicalPriceFeed.read.callAsync(
+        subjectDataDays,
+        { gas: DEFAULT_GAS}
+      );
+    }
+
+    it('returns the correct array', async () => {
+      const actualDailyPriceOutput = await subject();
+
+      const expectedDailyPriceOutput = updatedPrices.slice(-subjectDataDays.toNumber()).reverse();
+
+      expect(JSON.stringify(actualDailyPriceOutput)).to.equal(JSON.stringify(expectedDailyPriceOutput));
+    });
+
+    describe('when querying more data than available', async () => {
+      beforeEach(async () => {
+        subjectDataDays = new BigNumber(22);
+      });
+
+      it('should revert', async () => {
+        await expectRevertError(subject());
+      });
+    });
+  });
+
+  describe('#changeMedianizer', async () => {
+    let ethPrice: BigNumber;
+
+    let subjectNewMedianizer: Address;
+    let subjectCaller: Address;
+
+    beforeEach(async () => {
+      ethPrice = ether(150);
+      await oracleWrapper.updateMedianizerPriceAsync(
+        ethMedianizer,
+        ethPrice,
+        SetTestUtils.generateTimestamp(1000),
+      );
+
+      const updateFrequency = ONE_DAY_IN_SECONDS;
+      const updateTolerance = updateFrequency.div(4);
+      const medianizerAddress = ethMedianizer.address;
+      const dataDescription = '200DailyETHPrice';
```

Comment:
Put these as default values! Only create new constants here when you want to explicitly test the constructor.